troposphere/static/js/components/admin/ResourceRequest/ResourceRequest.jsx

```diff
@@ -36,148 +38,133 @@ export default React.createClass({
-            storeQuota.keys().some(attr => storeQuota.get(attr) !== quota[attr]);
-
-        let updatedQuota;
-        if (quotaChanged) {
-            updatedQuota =
-                // Search for quota in store based on attributes not id
-                quotas.findWhere(_.omit(quota, "id")) ||
-
-                // Updated quota doesn't exist, so create one
-                new Quota(_.omit(quota, "id"));
-        }
-
-        let changedAllocations = allocationSources
-            .filter(as => {
-                let storeAs = storeAllocations.get(as.uuid);
-                let allocationChanged =
-                    storeAs.keys().some(attr => storeAs.get(attr) !== as[attr])
-                return allocationChanged;
+        return Promise.resolve(allocationSource.save(allocationSource.pick("compute_allowed"), { patch: true }))
+            .then(() => {
+                Utils.dispatch(AllocationSourceConstants.UPDATE, {
+                    allocation: allocationSource,
+                    username: request.get("created_by").username
+                })
             })
-            .map(as => {
-                let storeAs = storeAllocations.get(as.uuid);
-                return storeAs.set(as);
+            .catch(errorHandler)
+    },
+
+    onIdentitySave(identity) {
+        let { selectedRequest: request } = this.props;
+        let quota = new Quota(_.omit(identity.get('quota'), ["id", "uuid"]));
+        return Promise.resolve(quota.save())
+            .then(
+                () => Promise.resolve(identity.save({ 'quota': quota.pick("id") }, { patch: true }))
+            )
+            .then(() => {
+                Utils.dispatch(IdentityConstants.UPDATE, {
+                    identity,
+                    username: request.get("created_by").username
+                });
             })
+            .catch(errorHandler)
+    },
 
+    onApprove() {
+        let { selectedRequest: request } = this.props;
+        let { statuses } = this.fetch();
         let status = statuses.findWhere({
             name: "approved"
         });
 
         this.setState({ actionPending: true });
-        ResourceActions.approve({
-            allocationSources: changedAllocations,
-            identity,
-            quota: updatedQuota,
-            status,
-            request,
-            response
-        }).then(() => {
-            browserHistory.push("/application/admin/resource-requests");
-        }).catch(this.setActionResolved)
+        ResourceRequestActions.updateRequest(request, status)
+            .then(() => {
+                browserHistory.push("/application/admin/resource-requests");
+            })
+            .always(this.setActionResolved);
     },
 
     setActionResolved() {
-        this.setState({ actionPending: false });
+        if (this.state.mounted) {
+            this.setState({ actionPending: false });
+        }
     },
 
-    onDeny(view) {
+    onDeny(reason) {
```

Comment:
It does not appear that `reason` is used. Curious if that is intentional (it likely is).

Reply:
welp that is a bug, good eye

Reply:
👍